python_modules/dagster/dagster/_core/definitions/asset_layer.py

```diff
@@ -636,6 +636,13 @@ def node_output_handle_for_asset(self, asset_key: AssetKey) -> NodeOutputHandle:
         check.invariant(len(matching_handles) == 1)
         return matching_handles[0]
 
+    def observable_asset_key_for_node(self, node_handle: NodeHandle) -> Optional[AssetKey]:
+        result = self.asset_info_by_node_output_handle.get(NodeOutputHandle(node_handle, "result"))
```

Comment:
its not obvious to me why this is the right look-up strategy for observable source assets - im guessing this is tied to some specific implementation detail?